ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -463,7 +464,29 @@ public static Path getCopyDestination(ReplChangeManager.FileInfo fileInfo, Path
     String[] subDirs = fileInfo.getSubDir().split(Path.SEPARATOR);
     Path destination = destRoot;
     for (String subDir: subDirs) {
-      destination = new Path(destination, subDir);
+      // If the directory is created by compactor, then the directory will have the transaction id also.
+      // In case of replication, the same txn id can not be used at target, as the txn with same id might be a
+      // aborted or live txn at target.
+      // In case of bootstrap load, we copy only the committed data, so the directory with only write id
+      // can be created. The validity txn id can be removed from the directory name.
+      // TODO : Support for incremental load flow. This can be done once replication of compaction is decided.
+      if (AcidUtils.getVisibilityTxnId(subDir) > 0) {
+        if (subDir.startsWith(AcidUtils.BASE_PREFIX)) {
+          AcidUtils.ParsedBase pb = AcidUtils.ParsedBase.parseBase(new Path(subDir));
+          destination = new Path(destination, AcidUtils.baseDir(pb.getWriteId()));
+        } else if (subDir.startsWith(AcidUtils.DELTA_PREFIX)) {
+          AcidUtils.ParsedDeltaLight pdl = AcidUtils.ParsedDeltaLight.parse(new Path(subDir));
+          destination = new Path(destination, AcidUtils.deltaSubdir(pdl.getMinWriteId(), pdl.getMaxWriteId()));
+        } else if (subDir.startsWith(AcidUtils.DELETE_DELTA_PREFIX)) {
+          AcidUtils.ParsedDeltaLight pdl = AcidUtils.ParsedDeltaLight.parse(new Path(subDir));
+          destination = new Path(destination, AcidUtils.deleteDeltaSubdir(pdl.getMinWriteId(), pdl.getMaxWriteId()));
```

Comment:
We are ignoring the statement id, I think because it's -1 always. Is that future proof?

Reply:
As of now ..statement id is not used any where in the code...and in replication flow we always ignore it.

Reply:
Hmm, when the statement ids are actually used, this might become a problem. For now this looks ok.